### PR TITLE
[GEOS-10545] Layer Group cache not initialized

### PR DESCRIPTION
**Pull request by ahmedababnehgeo**
[GEOS-10545](https://osgeo-org.atlassian.net/browse/GEOS-10545)

This PR resolves a cache issue that impacts the layer groups cache. We make sure that we call buildLayerGroupCache using a post processor bean. The bean checks whether the bean is of the desired type in order to build the cache for that bean (i.e. SecureCatalog object).
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by ahmedababnehgeo** [comment]
Old relevant PR (closed): https://github.com/geoserver/geoserver/pull/5964

**Comment by ahmedababnehgeo** [comment]
@taba90 PR ready for review

File: src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java [hunk 1]
```diff
@@ -119,7 +120,10 @@ public DefaultResourceAccessManager(DataAccessRuleDAO dao, Catalog rawCatalog) {
         this.dao = dao;
         this.rawCatalog = rawCatalog;
         this.root = buildAuthorizationTree(dao);
-        this.groupsCache = new LayerGroupContainmentCache(rawCatalog);
+    }
+
+    public void setGroupsCache(LayerGroupContainmentCache groupsCache) {
```
Comment (taba90) [inline src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java:125]: missing java doc on public method

File: src/extension/geofence/src/main/java/org/geoserver/geofence/GeofenceAccessManager.java [hunk 2]
```diff
@@ -134,6 +134,10 @@ public GeofenceAccessManager(
         this.helper = new GeoFenceAreaHelper();
     }
 
+    public void setGroupsCache(LayerGroupContainmentCache groupsCache) {
```
Comment (taba90) [inline src/extension/geofence/src/main/java/org/geoserver/geofence/GeofenceAccessManager.java:137]: missing java doc on public method

File: src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java [hunk 3]
```diff
@@ -674,4 +679,11 @@ public DataAccessLimits getAccessLimits(Authentication user, LayerInfo layer) {
     public LayerGroupAccessLimits getAccessLimits(Authentication user, LayerGroupInfo layerGroup) {
         return getAccessLimits(user, layerGroup, Collections.emptyList());
     }
+
+    protected LayerGroupContainmentCache getLayerGroupsCache() {
```
Comment (taba90) [inline src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java:683]: Missing java doc.

**Review by taba90 — COMMENTED** [review]
Looks good to me. Only few java docs missing on public/protected methods.

**Comment by ahmedababnehgeo** [comment]
PR updated. Java docs has been added.

**Review by taba90 — APPROVED** [review]
thanks looks good to me

**Pull request merged**